packages/unified/src/utilities/getNoteDependencies.ts

```diff
@@ -0,0 +1,82 @@
+import { DNodeCompositeKey } from "@dendronhq/common-all";
+import { Node, Data } from "unist";
+import visit from "unist-util-visit";
+import {
+  DendronASTTypes,
+  WikiLinkNoteV4,
+  NoteRefNoteV4,
+  HashTag,
+  UserTag,
+} from "../types";
+
+/**
+ * For a given AST, find all note dependencies whose data will be needed for
+ * rendering.
+ * @param ast the syntax tree to look for dependencies
+ * @returns an array of fname-vault? combinations that this tree depends on.
+ */
+export function getNoteDependencies(ast: Node<Data>): DNodeCompositeKey[] {
+  const renderDependencies: DNodeCompositeKey[] = [];
```

Comment:
do we need to dedup if we push the same keys?

Reply:
yes - I've actually done that in the next set of changes. I'll defer that change until the next PR to reduce conflicts if that's ok.